integration-tests/src/test/java/org/apache/druid/tests/indexer/AbstractStreamIndexingTest.java

```diff
@@ -67,92 +92,147 @@
   private IntegrationTestingConfig config;
 
   private StreamAdminClient streamAdminClient;
-  private WikipediaStreamEventStreamGenerator wikipediaStreamEventGenerator;
 
   abstract StreamAdminClient createStreamAdminClient(IntegrationTestingConfig config) throws Exception;
-  abstract StreamEventWriter createStreamEventWriter(IntegrationTestingConfig config) throws Exception;
-  abstract Function<String, String> generateStreamIngestionPropsTransform(String streamName,
-                                                                          String fullDatasourceName,
-                                                                          IntegrationTestingConfig config);
+
+  abstract StreamEventWriter createStreamEventWriter(IntegrationTestingConfig config, boolean transactionEnabled)
```

Comment:
Just to clarify on my earlier point. This should not have boolean transactionEnabled as a parameter. transactionEnabled 
 is a Kafka specific parameter and should be in the Kafka layer (AbstractKafkaIndexingServiceTest). This layer (AbstractStreamIndexingTest) should just get a eventWriter and write using the writer.

Reply:
Hmm, looking at again, I'm not sure how this can be implemented. Even though only `KafkaEventWriter` needs `transactionEnabled`, `createStreamEventWriter()` is called in `AbstractStreamIndexingTest` which is a super class of both `AbstractKafkaIndexingServiceTest` and `AbstractKinesisIndexingServiceTest`. That means, `AbstractStreamIndexingTest` needs to know about `transactionEnabled` parameter when it calls `createStreamEventWriter()`. Before this PR, you added `isKafkaWriterTransactionalEnabled()` to `AbstractKafkaIndexingServiceTest`, but this seems working only when all child classes of `AbstractKafkaIndexingServiceTest` run with only either transaction enabled or disabled which is not true in `ITKafkaIndexingServiceDataFormatTest` anymore. I prefer to parameterize this class to run with different transaction mode since it's better for code maintenance (we don't have to keep two separate test classes for each mode in sync).

A simple workaround I can come up with is adding a javadoc for this method explaining `transactionEnabled` parameter works for only Kafka. I assume this is ok since the `StreamEventWriter` interface has methods `isTransactionEnabled`, `initTransaction`, and `commitTransaction` which are implemented only for Kafka properly.

Reply:
hmm. I see. We can keep it this way then. My concern is that in Kinesis test class we are passing arbitrary meaningless true/false which does not do anything and might be misleading. I think Javadoc on the Kinesis test classes (ITKinesisIndexingServiceSerializedTest, etc) will definitely help. Another idea is to make boolean into Boolean and have the Kinesis test classes pass null instead of true/false to be more explicit that this transactionEnabled flag does not apply to Kinesis.

Reply:
Good idea. Raised https://github.com/apache/druid/pull/9795.